### DISCUSSION TITLE
OpenAPI spec of OpenID

### DISCUSSION THREAD

**Discussion by rhtenhove** [panva/openid-client#508] [Q&A]
We are looking for a way to interact with the OpenID provider based on all the endpoints it has available, the possible methods for each, and the content each request may have (such as query or body content).

The goal is security testing of an implementation of the provider.

The documentation of OpenID is already thorough, but there seems to be no machine readable reference of the spec. Is there something like this available?
It would be best if that were in an OpenAPI spec already :smile: 

Thanks!

/cc @jcnauta

**Comment by panva** [comment]
I don't see what OpenAPI spec has to do with the openid-client Node.js module.

**Reply by rhtenhove** [reply]
This client implements the specification of OpenID in node.js. As such it knows about most/all endpoints of an OpenID provider, and how these should be interacted with.

This looked like the place where knowledge of such a spec's existence would be.

Even if it isn't OpenAPI/Swagger, is there an overview of all endpoints and its allowed methods/components that an OpenID provider _usually_ provides?

**Reply by panva** [reply]
I am unsure whether I can suggest anything beyond referring you to the specifications or the code.
